docs/sidecar/scenarios/managed-identity.md

```diff
@@ -0,0 +1,347 @@
+
+# Grant admin consent
+az ad app permission admin-consent --id $IDENTITY_CLIENT_ID
+```
+
+### 4. Create Federated Identity Credential
+
+```bash
+# Create federated credential for Kubernetes service account
+az identity federated-credential create \
+  --name myapp-federated-identity \
+  --identity-name myapp-identity \
+  --resource-group myResourceGroup \
+  --issuer $AKS_OIDC_ISSUER \
+  --subject system:serviceaccount:default:myapp-sa
+```
+
+### 5. Create Kubernetes Service Account
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: myapp-sa
+  namespace: default
+  annotations:
+    azure.workload.identity/client-id: "<MANAGED_IDENTITY_CLIENT_ID>"
+```
+
+Apply:
+```bash
+kubectl apply -f serviceaccount.yaml
+```
+
+## Deployment Configuration
+
+### Complete Pod Configuration
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: myapp
+  namespace: default
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: myapp
+  template:
+    metadata:
+      labels:
+        app: myapp
+        azure.workload.identity/use: "true"  # Required for workload identity
+    spec:
+      serviceAccountName: myapp-sa
+      containers:
+      # Application container
+      - name: app
+        image: myregistry/myapp:latest
+        ports:
+        - containerPort: 8080
+        env:
+        - name: SIDECAR_URL
+          value: "http://localhost:5000"
+        resources:
+          requests:
+            memory: "256Mi"
+            cpu: "250m"
+          limits:
+            memory: "512Mi"
+            cpu: "500m"
+      
+      # Sidecar container
+      - name: sidecar
+        image: mcr.microsoft.com/identity/sidecar:latest
+        ports:
+        - containerPort: 5000
+        env:
+        # Azure AD Configuration
+        - name: AzureAd__Instance
+          value: "https://login.microsoftonline.com/"
+        - name: AzureAd__TenantId
+          value: "common"  # Or specific tenant ID
+        - name: AzureAd__ClientId
+          value: "<MANAGED_IDENTITY_CLIENT_ID>"
+        # No client secret or certificate needed!
+        
+        # Downstream API Configuration
+        - name: DownstreamApis__Graph__BaseUrl
+          value: "https://graph.microsoft.com/v1.0"
+        - name: DownstreamApis__Graph__Scopes
+          value: "User.Read Mail.Read"
+        
+        # Logging
+        - name: Logging__LogLevel__Default
+          value: "Information"
+        - name: Logging__LogLevel__Microsoft.Identity.Web
+          value: "Information"
+        
+        resources:
+          requests:
+            memory: "128Mi"
+            cpu: "100m"
+          limits:
+            memory: "256Mi"
+            cpu: "250m"
+        
+        livenessProbe:
+          httpGet:
+            path: /health
+            port: 5000
+          initialDelaySeconds: 10
+          periodSeconds: 10
+        
+        readinessProbe:
+          httpGet:
+            path: /health
```

Comment:
```suggestion
            path: /healthz
            port: 5000
          initialDelaySeconds: 10
          periodSeconds: 10
        
        readinessProbe:
          httpGet:
            path: /healthz
```

the container uses /healthz for this (ending in z is an accepted way to mark the endpoint not as part of the regular application logic)